embedded-database-spring-test/src/main/java/io/zonky/test/db/flyway/OptimizedFlywayTestExecutionListener.java

```diff
@@ -218,30 +219,55 @@ protected static MigrationVersion findFirstVersion(Flyway flyway, String... loca
     }
 
     protected static MigrationVersion findLastVersion(Flyway flyway, String... locations) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
-        MigrationResolver resolver = createMigrationResolver(flyway, locations);
-        Collection<ResolvedMigration> migrations = resolver.resolveMigrations();
-
+        Collection<ResolvedMigration> migrations = findMigrations(flyway, locations);
         if (CollectionUtils.isEmpty(migrations)) {
             return MigrationVersion.EMPTY;
         } else {
             return Iterables.getLast(migrations).getVersion();
         }
     }
 
+    protected static Collection<ResolvedMigration> findMigrations(Flyway flyway, String... locations) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        MigrationResolver resolver = createMigrationResolver(flyway, locations);
+        Collection<ResolvedMigration> migrations;
+        if (flywayVersion >= 52) {
+            ClassicConfiguration configInstance = (ClassicConfiguration) getField(flyway, "configuration");
+            migrations = invokeMethod(resolver, "resolveMigrations", new Context() {
+                @Override
+                public Configuration getConfiguration() {
+                    return configInstance;
+                }
+            });
+        } else {
+            migrations = invokeMethod(resolver, "resolveMigrations");
+        }
+
+        return migrations;
+    }
+
     protected static MigrationResolver createMigrationResolver(Flyway flyway, String... locations) throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
         String[] oldLocations = getFlywayLocations(flyway);
         try {
             flyway.setLocations(locations);
 
-            if (flywayVersion >= 51) {
+            if (flywayVersion >= 52) {
+                ClassLoader classLoader = OptimizedFlywayTestExecutionListener.class.getClassLoader();
+                ClassicConfiguration configInstance = (ClassicConfiguration) getField(flyway, "configuration");
```

Comment:
I would like the project to be compiled with Flyway 5.x.x version for now, so I've rewritten your code by using reflection.